_code-samples/modular-tutorials/account-support.js

```diff
@@ -0,0 +1,158 @@
+
+async function getAccount() {
+  let net = getNet()
+  const client = new xrpl.Client(net)
+  await client.connect()
+  let results = `\nConnected to ${net}.`
+  let faucetHost = null
+  const my_wallet = (await client.fundWallet(null, { faucetHost})).wallet
+  const newAccount = [my_wallet.address, my_wallet.seed]
+  return (newAccount)
+  client.disconnect()
+} // End of getAccount()
+
+async function getNewAccount1() {
+  account1address.value = "Getting new account."
+  const accountInfo= await getAccount()
+  account1address.value = accountInfo[0]
+  account1seed.value = accountInfo[1]
+}
+
+async function getNewAccount2() {
+  account2address.value = "Getting new account."
+  const accountInfo= await getAccount()
+  account2address.value = accountInfo[0]
+  account2seed.value = accountInfo[1]
+}
+
+// *****************************************************
+// ********** Get Account from Seed ******************** 
+// *****************************************************
+
+async function getAccountFromSeed(my_seed) {
+  const net = getNet()
+  const client = new xrpl.Client(net)
+  await client.connect()
+  let results = '\nConnected, finding wallet.\n'
+  resultField.value = results
+  const wallet = xrpl.Wallet.fromSeed(my_seed)
+// ----------------------Populate the fields for left and right accounts.
+  const address = wallet.address
+  client.disconnect()
+  return (address)
+} // End of getAccountFromSeed()
+
+// *****************************************************
+// ********** Get Account from Seed1 ******************* 
+// *****************************************************
+
+async function getAccountFromSeed1() {
+  account1address.value = await getAccountFromSeed(account1seed.value)
+}
+
+// *****************************************************
+// ********** Get Account from Seed2 ******************* 
+// *****************************************************
+
+async function getAccountFromSeed2() {
+  account2address.value = await getAccountFromSeed(account2seed.value)
+}
+
+// *****************************************************
+// ************ Gather Account Info ********************
+// *****************************************************
+
+function gatherAccountInfo() {
+  let accountData = account1name.value + "\n" + account1address.value + "\n" + account1seed.value + "\n"
+  accountData += account2name.value + "\n" + account2address.value + "\n" + account2seed.value
+  resultField.value = accountData
+}
+
+// *****************************************************
+// ********** Distribute Account Info ******************
+// *****************************************************
+
+function distributeAccountInfo() {
+  let accountInfo = resultField.value.split("\n")
+  account1name.value = accountInfo[0]
+  account1address.value = accountInfo[1]
+  account1seed.value = accountInfo[2]
+  account2name.value = accountInfo[3]
+  account2address.value = accountInfo[4]
+  account2seed.value = accountInfo[5]
+}
```

Comment:
I'm seeing some weird behaviour in the `send-mpt.html` example when I click "Distribute Account Info" (which calls this function).

<img width="1285" alt="Screenshot 2025-04-25 at 12 05 57" src="https://github.com/user-attachments/assets/495dd283-78db-4c0f-8891-e5fdfe5debbe" />

It looks like the fields are not populated correctly.

I printed the contents of the `accountInfo` variable in the console to help see what's in it. Seems that we don't have a value for **name**, and **seed** is populated with the wrong value?

Reply:
The instructions are pretty clear in the doc. You paste the gathered info, cut and paste the MPT Issuance ID, click Distribute Account Info, then **Get New Account 2** or **Get Account 2 From Seed**. It's a little convenience tool to be able to gather and distribute info - if people choose not to use it, that's okay. But making it robust enough to distribute info in a number of formats is overkill for the tutorials.